File: src/main/java/it/alian/gun/mesmerize/compat/Dependency.java
package it.alian.gun.mesmerize.compat;

import it.alian.gun.mesmerize.MConfig;
import it.alian.gun.mesmerize.MLocale;
import it.alian.gun.mesmerize.Mesmerize;
import it.alian.gun.mesmerize.compat.hook.MesmerizeHolograph;
import it.alian.gun.mesmerize.compat.hook.MesmerizePlaceholder;
import it.alian.gun.mesmerize.compat.hook.MesmerizeSkillApi;
import org.bukkit.Bukkit;

public class Dependency {

    public static void init() {
        if (!Bukkit.getPluginManager().isPluginEnabled("PowerNBT")) {
            MLocale.WARN_DEPENDENCY_MISSING.console("PowerNBT");
            throw new RuntimeException(MLocale.WARN_DEPENDENCY_MISSING.msg("PowerNBT"));
        }
        Equipment.init();
        if (MConfig.General.useRPGInventory && Bukkit.getPluginManager().isPluginEnabled("RPGInventory")) {
            MLocale.GENERAL_HOOK.console("RPGInventory");
        }
        if (Bukkit.getPluginManager().isPluginEnabled("PlaceholderAPI")) {
            new MesmerizePlaceholder(Mesmerize.instance, "mesmerize").hook();
            MLocale.GENERAL_HOOK.console("PlaceholderAPI");
        }
        if (Bukkit.getPluginManager().isPluginEnabled("SkillAPI")) {
            MesmerizeSkillApi.init();
            MLocale.GENERAL_HOOK.console("SkillAPI");
        }
        MesmerizeHolograph.init();
        if (Bukkit.getPluginManager().isPluginEnabled("HolographicDisplays")) {
            MLocale.GENERAL_HOOK.console("HolographicDisplays");
        }
    }

}

File: src/main/scala/it/alian/gun/mesmerize/compat/Dependency.scala
package it.alian.gun.mesmerize.compat

import com.ilummc.tlib.resources.TLocale
import it.alian.gun.mesmerize.MesmerizeDelegate._
import it.alian.gun.mesmerize.compat.hook.{MesmerizePlaceholder, MesmerizeSkillApi, MesmerizeVault}
import it.alian.gun.mesmerize.scalaapi.Prelude._
import org.bukkit.Bukkit

object Dependency {
  def init(): Unit = {
    if (!Bukkit.getPluginManager.isPluginEnabled("PowerNBT")) {
      error("WARN_DEPENDENCY_MISSING", "PowerNBT")
      throw new RuntimeException(TLocale.asString("WARN_DEPENDENCY_MISSING", "PowerNBT"))
    }
    if (Bukkit.getPluginManager.isPluginEnabled("PlaceholderAPI")) {
      new MesmerizePlaceholder(instance, "mesmerize").hook
      info("GENERAL_HOOK", "PlaceholderAPI")
    }
    if (Bukkit.getPluginManager.isPluginEnabled("SkillAPI")) {
      MesmerizeSkillApi.init()
      info("GENERAL_HOOK", "SkillAPI")
    }
    if (Bukkit.getPluginManager.isPluginEnabled("HolographicDisplays"))
      info("GENERAL_HOOK", "HolographicDisplays")
    MesmerizeVault.init()
    if (Bukkit.getPluginManager.isPluginEnabled("Vault"))
      info("GENERAL_HOOK", "Vault")
  }
}